### ISSUE TITLE
Replace a String parameter documentation with the corresponding full type model?

### ISSUE THREAD

**Issue by watery** [springfox/springfox#2523]
- [x] What version of the library are you using?
2.9.2

What kind of issue is this?

 - [x] Question

I need to receive the raw JSON in a PATCH request body, since I'll use it to do a partial update on an existing object, but I want the documentation to anyway describe that parameter as the full object type, so that in

```
@PatchMapping(path="/{id}")
public ApiResponse<MyObject> patch(@PathVariable String id, @RequestBody String myObjText) {
```

`myObjText` will be documented as having a model of type `MyObject`.

I've already tried combining `@ApiParam(hidden = true)` and `@ApiImplicitParam`, but it doesn't seem to work.

Here's a link to [my SO question](https://stackoverflow.com/q/51098932/3127111) just in case.

**Comment by watery** [comment]
I found the [answer to FAQ 27](http://springfox.github.io/springfox/docs/snapshot/#q27) (referenced in [dilipkrish comment to #2352](https://github.com/springfox/springfox/issues/2352#issuecomment-380297386) ) and that's almost like my requirement.

I got it working: I was using `@ApiImplicitParam( ..., dataType="<package-name.class-name>")`, when I changed that to `datatype=<class-name>` the model gets rendered in the docs as I want.
If I instead employ the `dataTypeClass` parameter of `@ApiImplicitParam` the parameter is yet documented as a plain string.

Is this the expected usage?

**Comment by watery** [comment]
@dilipkrish can you help me on this?

**Comment by dilipkrish** [comment]
@watery Yes the dataTypeClass isnt supported yet. It seems like a new annotation property that was introduced.

**Comment by rahulaga** [comment]
Thanks @dilipkrish. I have 2 questions. 
1. In the definition it does look to add `$ref: '#/definitions/CustomTypeFor2031'` but it is actually missing in the definitions below. Similarly if I use a non-static class it shows correctly in the ref but does not actually get added to the definitions.
2. I have a further use case where I want to `directModelSubstitute` the class `CustomTypeFor2031` with something else. That does not look to work either.

I would appreciate any guidance.

**Comment by dilipkrish** [comment]
@rahulaga You need to add that type to the additional models in the docket.

Regarding your second question, its kind of awkward what you're trying to do here if you think about it. You're saying "Im modeling this interaction with `CustomTypeFor2031` but I dont want you to use that type, I want you to use this other type". Why not tell it to use the other type?

**Comment by rahulaga** [comment]
Ok thanks. 
Yes, I am creating multiple dockets so want to do different substitutions in the different dockets. I'm using `alternateTypeRules` for now.
